pkg/virt-controller/services/template.go

```diff
@@ -303,7 +303,7 @@ func (t *templateService) RenderLaunchManifest(vmi *v1.VirtualMachineInstance) (
 
 	containers = append(containers, container)
 
-	hostName := vmi.Name
+	hostName := strings.Split(vmi.Name, ".")[0]
```

Comment:
I think that we really should have one common helper function between cloud-init and pod.